diff --git a/snappymail/v/2.9.5/app/libraries/MailSo/Base/Utils.php b/snappymail/v/2.9.5/app/libraries/MailSo/Base/Utils.php
index 15b78e494..0aaff764e 100644
--- a/snappymail/v/2.9.5/app/libraries/MailSo/Base/Utils.php
+++ b/snappymail/v/2.9.5/app/libraries/MailSo/Base/Utils.php
@@ -1273,7 +1273,7 @@ public static function FunctionExistsAndEnabled($mFunctionNameOrNames) : bool
 
 		return !empty($mFunctionNameOrNames)
 			&& \function_exists($mFunctionNameOrNames)
-			&& !\is_callable($mFunctionNameOrNames);
+			&& \is_callable($mFunctionNameOrNames);
 	}
 
 	public static function ClearNullBite($mValue) : string
diff --git a/snappymail/v/2.9.5/app/libraries/MailSo/Imap/ImapClient.php b/snappymail/v/2.9.5/app/libraries/MailSo/Imap/ImapClient.php
index 99ca8ea0f..628adbc8f 100644
--- a/snappymail/v/2.9.5/app/libraries/MailSo/Imap/ImapClient.php
+++ b/snappymail/v/2.9.5/app/libraries/MailSo/Imap/ImapClient.php
@@ -98,8 +98,7 @@ public function Connect(string $sServerName, int $iPort = 143,
 
 		$this->aCapabilityItems = $this->getResponse('*')->getCapabilityResult();
 
-		if (\MailSo\Net\Enumerations\ConnectionSecurityType::UseStartTLS(
-			$this->IsSupported('STARTTLS'), $this->iSecurityType))
+		if ($this->IsSupported('STARTTLS') && \MailSo\Net\Enumerations\ConnectionSecurityType::UseStartTLS($this->iSecurityType))
 		{
 			$this->SendRequestGetResponse('STARTTLS');
 			$this->EnableCrypto();
diff --git a/snappymail/v/2.9.5/app/libraries/MailSo/Net/Enumerations/ConnectionSecurityType.php b/snappymail/v/2.9.5/app/libraries/MailSo/Net/Enumerations/ConnectionSecurityType.php
index 485eedeef..317fdf693 100644
--- a/snappymail/v/2.9.5/app/libraries/MailSo/Net/Enumerations/ConnectionSecurityType.php
+++ b/snappymail/v/2.9.5/app/libraries/MailSo/Net/Enumerations/ConnectionSecurityType.php
@@ -47,11 +47,11 @@ public static function UseSSL(int $iPort, int $iSecurityType) : bool
 		return self::SSL === $iResult;
 	}
 
-	public static function UseStartTLS(bool $bSupported, int $iSecurityType, bool $bHasSupportedAuth = true) : bool
+	public static function UseStartTLS(int $iSecurityType, bool $bHasSupportedAuth = true) : bool
 	{
-		return ($bSupported &&
-			(self::STARTTLS === $iSecurityType ||
-				(self::AUTO_DETECT === $iSecurityType && (!$bHasSupportedAuth || \MailSo\Config::$PreferStartTlsIfAutoDetect))) &&
-			\defined('STREAM_CRYPTO_METHOD_TLS_CLIENT') && \MailSo\Base\Utils::FunctionExistsAndEnabled('stream_socket_enable_crypto'));
+		return
+			(self::STARTTLS === $iSecurityType
+			 || (self::AUTO_DETECT === $iSecurityType && (!$bHasSupportedAuth || \MailSo\Config::$PreferStartTlsIfAutoDetect)))
+		 && \defined('STREAM_CRYPTO_METHOD_TLS_CLIENT') && \MailSo\Base\Utils::FunctionExistsAndEnabled('stream_socket_enable_crypto');
 	}
 }
diff --git a/snappymail/v/2.9.5/app/libraries/MailSo/Sieve/ManageSieveClient.php b/snappymail/v/2.9.5/app/libraries/MailSo/Sieve/ManageSieveClient.php
index 3f9690c26..24c30a28a 100644
--- a/snappymail/v/2.9.5/app/libraries/MailSo/Sieve/ManageSieveClient.php
+++ b/snappymail/v/2.9.5/app/libraries/MailSo/Sieve/ManageSieveClient.php
@@ -72,8 +72,7 @@ public function Connect(string $sServerName, int $iPort,
 		$this->validateResponse($aResponse);
 		$this->parseStartupResponse($aResponse);
 
-		if (\MailSo\Net\Enumerations\ConnectionSecurityType::UseStartTLS(
-			$this->IsSupported('STARTTLS'), $this->iSecurityType))
+		if ($this->IsSupported('STARTTLS') && \MailSo\Net\Enumerations\ConnectionSecurityType::UseStartTLS($this->iSecurityType))
 		{
 			$this->sendRequestWithCheck('STARTTLS');
 			$this->EnableCrypto();
diff --git a/snappymail/v/2.9.5/app/libraries/MailSo/Smtp/SmtpClient.php b/snappymail/v/2.9.5/app/libraries/MailSo/Smtp/SmtpClient.php
index e19125dd6..cc1804c61 100644
--- a/snappymail/v/2.9.5/app/libraries/MailSo/Smtp/SmtpClient.php
+++ b/snappymail/v/2.9.5/app/libraries/MailSo/Smtp/SmtpClient.php
@@ -432,8 +432,7 @@ private function preLoginStartTLSAndEhloProcess(string $sEhloHost) : void
 
 		$this->ehloOrHelo($sEhloHost);
 
-		if (\MailSo\Net\Enumerations\ConnectionSecurityType::UseStartTLS(
-			$this->IsSupported('STARTTLS'), $this->iSecurityType, $this->HasSupportedAuth()))
+		if ($this->IsSupported('STARTTLS') && \MailSo\Net\Enumerations\ConnectionSecurityType::UseStartTLS($this->iSecurityType, $this->HasSupportedAuth()))
 		{
 			$this->sendRequestWithCheck('STARTTLS', 220);
 			$this->EnableCrypto();
